sycl/tools/sycl-ls/sycl-ls.cpp

```diff
@@ -106,14 +102,26 @@ int main(int argc, char **argv) {
     return EXIT_FAILURE;
   }
 
+  const char *filter = std::getenv("SYCL_DEVICE_FILTER");
```

Comment:
It is better to use SYCLConfig here because the value can be set either in environment variable or in sycl.conf
Wording for warning should be also changed.

Reply:
You probably already know that sycl-ls is a SYCL app, and it is not a part of SYCL RT.
Do you think it is better to use SYCL internal data structure at the user application level?